# Crossmatch habitable zone targets with the CKS sample, Gaia DR2 and my own rotation period catalogue.

In [11]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import pandas as pd
from astropy.io import fits
from astropy.table import Table
import chronology as ch

Load Stephen Kane's list of targets, the CKS catalogue, the McQuillan catalogue, my own rotation period catalogue and Gaia DR2.

In [4]:
# Kane catalogue:
kane = pd.read_csv("../data/kane_targets.csv")

# CKS catalogue:
cks = pd.read_csv("../data/cks_physical_merged.csv")

# McQuillan catalogue:
mc = pd.read_csv("../data/Table_1_Periodic.txt")

# Tim and my catalogue:
tdmra = pd.read_csv("../data/koi_periods_0712.csv")

# Gaia DR2
dat = Table.read('../data/kepler_dr2_4arcsec_updated.fits', format='fits')
dr2 = dat.to_pandas()

/Users/ruthangus/anaconda/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2881: TableReplaceWarning: converted column 'teff_err1' from integer to float
  exec(code_obj, self.user_global_ns, self.user_ns)
/Users/ruthangus/anaconda/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2881: TableReplaceWarning: converted column 'teff_err2' from integer to float
  exec(code_obj, self.user_global_ns, self.user_ns)


What's the overlap between Stephen Kane's catalogue and the CKS sample?

In [5]:
# Format the CKS catalogue and Kane catalogue column headers.
koi_ids = []
for i in cks.id_koicand:
    koi_ids.append("Kepler-{}".format(int(float(i[1:]))))
cks["koi_id"] = koi_ids

KOIs = np.zeros(np.shape(kane)[0], dtype=int)
for j, i in enumerate(kane.koi_id):
    if i[:7] == "Kepler-":
        KOIs[j] = int((float(i[7:])))

kane["KOI"] = KOIs
print(kane.keys(), "\n")
print(cks.keys())
kane_cks = pd.merge(kane, cks, on="koi_id", how="inner", suffixes=["_KIC", "_CKS"])
print(np.shape(kane), np.shape(cks), np.shape(kane_cks))

Index(['koi_id', 'planet_designation', 'p_orb_days', 'a_AU', 'r_pl_earth',
       'teff', 'r_star_rsol', 'insolation_flux_F_E', 'KOI'],
      dtype='object') 

Index(['id_starname', 'id_kic', 'id_koicand', 'id_kepler_name',
       'koi_disposition', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0', 'koi_time0_err1',
       ...
       'iso_sma', 'iso_sma_err1', 'iso_sma_err2', 'iso_insol',
       'iso_insol_err1', 'iso_insol_err2', 'iso_teq', 'iso_teq_err1',
       'iso_teq_err2', 'koi_id'],
      dtype='object', length=121)
(1162, 9) (2025, 121) (617, 129)


617 stars have CKS spectra. ,How many of these stars have rotation periods measured by Amy McQuillan?

In [6]:
kane_cks_mc = pd.merge(kane_cks, mc, left_on="id_kic", right_on="kepid", how="inner", suffixes=["", "_mc"])
np.shape(kane_cks_mc)

(0, 140)

None... How about in Tim and my catalogue?

In [10]:
print(tdmra.keys())

Index(['Unnamed: 0', 'KOI', 'feh', 'feh_errm', 'feh_errp', 'logg', 'logg_errm',
       'logg_errp', 'period', 'period_errm', 'period_errp', 'teff',
       'teff_errm', 'teff_errp'],
      dtype='object')


In [7]:
kane_cks_tdmra = pd.merge(kane_cks, tdmra, on="KOI", how="inner", suffixes=["", "_tdmra"])
np.shape(kane_cks_tdmra)

(552, 142)

How many of these have Gaia DR2 distances?

In [8]:
# Make cuts on parallax signal-to-noise
m = 10 < dr2.parallax_over_error.values  # Signal to noise greater than 10
gdr2 = dr2.iloc[m]

gdr2.kepid.values[0]
int_kepid = []
for i in gdr2.kepid.values:
    int_kepid.append(int(i))
gdr2["int_kepid"] = np.array(int_kepid, dtype=int)

kane_cks_tdmra_dr2 = pd.merge(kane_cks_tdmra, gdr2, left_on="id_kic", right_on="int_kepid", how="inner")
print(np.shape(kane_cks_tdmra_dr2))

kane_cks_tdmra_dr2.to_csv("kane_cks_tdmra_dr2.csv")

/Users/ruthangus/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


(546, 244)


So there are 546 stars with spectroscopic properties, distances AND rotation periods. The next thing to do is to combine all this information using isochrones and a gyrochronology model.

In [9]:
df = pd.read_csv("kane_cks_tdmra_dr2.csv")
for k in df.keys():
    print(k)

Unnamed: 0
koi_id
planet_designation
p_orb_days
a_AU
r_pl_earth
teff_x
r_star_rsol
insolation_flux_F_E
KOI
id_starname
id_kic
id_koicand
id_kepler_name
koi_disposition
koi_period
koi_period_err1
koi_period_err2
koi_time0
koi_time0_err1
koi_time0_err2
koi_impact
koi_impact_err1
koi_impact_err2
koi_duration
koi_duration_err1
koi_duration_err2
koi_ingress
koi_ingress_err1
koi_ingress_err2
koi_depth
koi_depth_err1
koi_depth_err2
koi_ror
koi_ror_err1
koi_ror_err2
koi_srho
koi_srho_err1
koi_srho_err2
koi_prad
koi_prad_err1
koi_prad_err2
koi_sma
koi_sma_err1
koi_sma_err2
koi_teq
koi_teq_err1
koi_teq_err2
koi_insol
koi_insol_err1
koi_insol_err2
koi_dor
koi_dor_err1
koi_dor_err2
koi_max_mult_ev
koi_model_snr
koi_count
koi_num_transits
koi_steff
koi_steff_err1
koi_steff_err2
koi_slogg
koi_slogg_err1
koi_slogg_err2
koi_smet
koi_smet_err1
koi_smet_err2
koi_srad
koi_srad_err1
koi_srad_err2
koi_smass
koi_smass_err1
koi_smass_err2
koi_sage
koi_sage_err1
koi_sage_err2
koi_sparprov
kic_kepmag
kic_jmag


In [25]:
savedir = "planet_results"

prot, prot_err = df.period, df.period_errm

N = len(df)
for i, id in enumerate(df.id_kic.values):
    print(i, "of", N)
    print("kepid = ", str(id).zfill(9))
    
    iso_params = {"G": (df.phot_g_mean_mag[i], df.phot_g_mean_mag[i]*.05),  # 5% uncertainties on photometry.
                  "bp": (df.phot_bp_mean_mag[i], df.phot_bp_mean_mag[i]*.05),
                  "rp": (df.phot_rp_mean_mag[i], df.phot_rp_mean_mag[i]*.05),
                  "J": (df.kic_jmag[i], df.kic_jmag[i]*.05),
                  "H": (df.kic_hmag[i], df.kic_hmag[i]*.05),
                  "K": (df.kic_kmag[i], df.kic_kmag[i]*.05),
                  "teff": (df.iso_steff[i], df.iso_steff_err2[i]),
                  "logg": (df.iso_slogg[i], df.iso_slogg_err2[i]),
                  "feh": (df.iso_smet[i], df.iso_smet_err2[i]),
                  "parallax": (df.parallax[i], df.parallax_error[i])} 
    
    st = ch.star(iso_params, prot[i], prot_err[i], savedir=savedir, suffix=str(id).zfill(9))
    sampler = st.fit()

    st.make_plots()

0 of 546
kepid =  000757450


  0%|          | 74/100000 [00:01<22:58, 72.51it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [1:00:14<00:00, 27.66it/s]


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
1 of 546
kepid =  006029239


  0%|          | 71/100000 [00:01<24:17, 68.57it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 62%|██████▏   | 62000/100000 [33:14<20:22, 31.08it/s]    


nsteps =  62000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
2 of 546
kepid =  006029239


  0%|          | 68/100000 [00:00<23:21, 71.30it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 62%|██████▏   | 62000/100000 [25:32<15:39, 40.44it/s]  


nsteps =  62000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
3 of 546
kepid =  008041216


  0%|          | 71/100000 [00:01<24:40, 67.48it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 22%|██▏       | 22100/100000 [06:34<18:28, 70.28it/s]  


nsteps =  22100 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
4 of 546
kepid =  012105051


  0%|          | 71/100000 [00:01<24:42, 67.41it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [15:27:43<00:00, 62.90it/s]      


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
5 of 546
kepid =  012105051


  0%|          | 71/100000 [00:01<24:07, 69.02it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [2:07:22<00:00, 13.08it/s]    


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
6 of 546
kepid =  004180280


  0%|          | 68/100000 [00:00<22:56, 72.60it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [1:26:27<00:00, 19.28it/s] 


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...


7 of 546
kepid =  004180280


  0%|          | 67/100000 [00:00<23:42, 70.26it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [1:01:04<00:00, 27.29it/s]


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...


8 of 546
kepid =  005735762


  0%|          | 67/100000 [00:00<23:58, 69.48it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [58:47<00:00, 28.35it/s] 


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...


9 of 546
kepid =  005735762


  0%|          | 69/100000 [00:00<22:40, 73.48it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [59:19<00:00, 28.09it/s] 


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...


10 of 546
kepid =  005735762


  0%|          | 73/100000 [00:01<24:08, 68.97it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [57:00<00:00, 29.24it/s] 


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...


11 of 546
kepid =  003835670


  0%|          | 69/100000 [00:00<23:01, 72.34it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [59:01<00:00, 28.24it/s] 


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
12 of 546
kepid =  007626506


  0%|          | 74/100000 [00:01<23:30, 70.86it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 75%|███████▌  | 75300/100000 [42:09<13:49, 29.76it/s]   


nsteps =  75300 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
13 of 546
kepid =  007626506


  0%|          | 74/100000 [00:01<23:25, 71.10it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 75%|███████▌  | 75300/100000 [38:26<12:36, 32.64it/s]   


nsteps =  75300 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
14 of 546
kepid =  008247638


  0%|          | 74/100000 [00:01<28:45, 57.92it/s]  /Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 21%|██        | 21200/100000 [2:32:17<9:26:05,  2.32it/s]    


nsteps =  21200 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
15 of 546
kepid =  008247638


  0%|          | 69/100000 [00:01<25:00, 66.62it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 21%|██        | 21200/100000 [10:54:20<40:32:08,  1.85s/it]    


nsteps =  21200 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
16 of 546
kepid =  008247638


  0%|          | 73/100000 [00:01<24:31, 67.93it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 21%|██        | 21200/100000 [06:17<23:22, 56.20it/s]  


nsteps =  21200 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
17 of 546
kepid =  008247638


  0%|          | 72/100000 [00:01<23:50, 69.83it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 21%|██        | 21200/100000 [19:36<1:12:52, 18.02it/s]  


nsteps =  21200 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
18 of 546
kepid =  008672910


  0%|          | 70/100000 [00:01<25:03, 66.45it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [2:44:56<00:00, 59.84it/s]    


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
19 of 546
kepid =  008672910


  0%|          | 68/100000 [00:00<24:16, 68.61it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [1:00:47<00:00, 27.41it/s]


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
20 of 546
kepid =  003116412


  0%|          | 71/100000 [00:01<24:28, 68.06it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [1:19:06<00:00, 21.07it/s]  


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
21 of 546
kepid =  003114811


  0%|          | 68/100000 [00:01<27:13, 61.17it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 18%|█▊        | 18500/100000 [05:23<23:44, 57.20it/s]  


nsteps =  18500 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
22 of 546
kepid =  006307537


  0%|          | 73/100000 [00:01<23:50, 69.88it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
100%|██████████| 100000/100000 [2:18:59<00:00, 11.99it/s]   


nsteps =  100000 burnin =  10000
Plotting age posterior
Plotting production chains...
Making corner plot...
23 of 546
kepid =  006359320


  0%|          | 71/100000 [00:01<26:11, 63.60it/s]/Users/ruthangus/anaconda/lib/python3.5/site-packages/isochrones-1.2.2.dev0-py3.5.egg/isochrones/starmodel.py:523: RuntimeWarning: divide by zero encountered in log
  mass_lnprior = np.log(self.prior('mass', masses[0]))
 61%|██████▏   | 61299/100000 [10:07:46<6:23:43,  1.68it/s]     


RuntimeError: Can't decrement id ref count (File write failed: time = thu oct 11 01:00:55 2018
, filename = 'planet_results/006359320_samples.h5', file descriptor = 50, errno = 28, error message = 'no space left on device', buf = 0x11ba44cd0, total write size = 3384, bytes this sub-write = 3384, bytes actually written = 18446744073709551615, offset = 510713856)